src/hooks/useGamification.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react";
+
+import { useGlobalState } from "../contexts/global";
+
+export const useGamification = () => {
+  const { points, setShowGamification } = useGlobalState();
+  const [isFirstRender, setIsFirstRender] = useState<boolean>(true);
```

Comment:
The first render is not the first time the user has ever loaded the content, so storing that the user has rendered in localStorage is not needed?